src/main/java/org/openrewrite/sql/FormatSql.java

```diff
@@ -0,0 +1,191 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.sql;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.SourceFile;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.style.IntelliJ;
+import org.openrewrite.java.style.TabsAndIndentsStyle;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import com.github.vertical_blank.sqlformatter.SqlFormatter;
+import com.github.vertical_blank.sqlformatter.languages.Dialect;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class FormatSql extends Recipe {
+    @Option(displayName = "SQL dialect to be used to format SQL snippets.", description = "Check out https://github.com/vertical-blank/sql-formatter#dialect for supported dialects.", valid = {"sql", "mariadb", "mysql", "postgresql", "db2", "plsql", "n1ql", "redshift", "spark", "tsql"}, example = "PostgreSql", required = false)
+    @Nullable
+    String sqlDialect;
+
+    public FormatSql() {
+        sqlDialect = Dialect.StandardSql.name();
+    }
+
+    public FormatSql(String sqlDialect) {
+        this.sqlDialect = sqlDialect;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Format SQL in String Text Blocks";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Checks whether a text block may contain SQL, and if so, formats the text accordingly.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesJavaVersion<>(15);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.Literal visitLiteral(J.Literal lit, ExecutionContext ctx) {
+                J.Literal literal = super.visitLiteral(lit, ctx);
+                if (isTextBlock(literal)) {
+                    String originalValue = (String) literal.getValue();
+                    if (new SqlDetector().isSql(originalValue)) {
+                        String formatted = SqlFormatter.of(Dialect.valueOf(sqlDialect)).format(originalValue);
+                        TabsAndIndentsStyle style = getCursor().firstEnclosingOrThrow(SourceFile.class)
+                                .getStyle(TabsAndIndentsStyle.class);
+                        String indented = indent(literal.getValueSource(), formatted, style);
+                        return literal
+                                .withValue(indented)
```

Comment:
The indentation should only be part of the `valueSource` not the `value`. This is a detail, but it could cause problems in other recipes if these two values don't match up.

```suggestion
                                .withValue(formatted)
```

Reply:
Yes I think I added that because I saw some cycle issues earlier; let's see if this broke the build and how to handle that then